test/auth-tests/src/it/java/software/amazon/awssdk/auth/source/UserAgentProviderTest.java

```diff
@@ -74,8 +74,8 @@ void userAgentString_containsCredentialProviderNames_IfPresent(IdentityProvider<
 
     private static Stream<Arguments> credentialProviders() {
         return Stream.of(
-            Arguments.of(StaticCredentialsProvider.create(SESSION_IDENTITY), "stat"),
-            Arguments.of(StaticCredentialsProvider.create(BASIC_IDENTITY), "stat")
+            Arguments.of(StaticCredentialsProvider.create(SESSION_IDENTITY), "m/D"),
+            Arguments.of(StaticCredentialsProvider.create(BASIC_IDENTITY), "m/D")
```

Comment:
Probably missed something, but why are we checking for `m/D` here? (`D` being `RETRY_MODE_LEGACY`).

Reply:
I think there is a pretty large bug in how the user agent is constructed, putting credential provider business metrics in auth source but not in `m`.